sql/rename.go

```diff
@@ -136,3 +136,57 @@ func (p *planner) RenameTable(n *parser.RenameTable) (planNode, error) {
 
 	return &valuesNode{}, nil
 }
+
+// RenameIndex renames the index.
+// Privileges: WRITE on database.
+//   Notes: postgres requires the table owner.
+//          mysql requires ALTER, CREATE, and INSERT on the table.
+func (p *planner) RenameIndex(n *parser.RenameIndex) (planNode, error) {
+	if n.NewName == "" {
+		return nil, errEmptyIndexName
+	}
+
+	if err := n.Name.NormalizeTableName(p.session.Database); err != nil {
+		return nil, err
+	}
+
+	if n.Name.Index() == string(n.NewName) {
+		// Noop.
```

Comment:
I wonder if it should be a Noop to rename an index to the same name even if the referenced database/table/index do not exist.